### PR TITLE
Update dev deps and add new mypy check

### PR DESCRIPTION
**Pull request by palango**
~~Merge after #496~~

Updates the dev dependencies and enables a new mypy check

### CODE REVIEW

File: src/monitoring_service/handlers.py [hunk 1]
```diff
@@ -490,8 +490,7 @@ def action_claim_reward_triggered_event_handler(event: Event, context: Context)
 
     # check that the latest update was ours and that we didn't send a transaction yet
     can_claim = (
-        channel is not None
-        and channel.claim_tx_hash is None
+        channel.claim_tx_hash is None
```
Comment (Dominik1999) [inline src/monitoring_service/handlers.py]: is this related to the title? Here I understand that we add a new condition to the `can_claim` statement

Comment (palango) [reply src/monitoring_service/handlers.py]: That's a "problem" found by a new option in the updated mypy, I adapted the title.
